api/src/main/java/org/apache/iceberg/expressions/PathUtil.java

```diff
@@ -61,4 +74,59 @@ static List<String> parse(String path) {
 
     return names;
   }
+
+  public static String toNormalizedPath(Iterable<String> fields) {
```

Comment:
This has to be public so that the Parquet conversion code can use it to translate a list of field names into a normalized path.